client/state/data-layer/freshness-verifier/README.md

```diff
@@ -0,0 +1,45 @@
+```js
+[ COMMENTS_REQUEST ]: [ dispatchRequest( fetchPostComments, addComments, announceFailure, noop, { freshness: 10000 } ) ],
+```
+The freshness value the data-layer sets acts as a default value if a data requester has a particular need for a different freshness value it can pass that value in the action being dispatched:
+```js
+{
+ type: COMMENTS_REQUEST,
+ siteId: 12886750,
+ postId: 97,
+ ...
+ freshness: 2000
+}
+```
+When a request is still fresh it does not trigger a network request and is ignored.
+
+The freshness mechanism is optional and if no freshness is set in the handler or the action the freshness mechanism is not executed. If a freshness value is set on the handler but in a given component freshness should not be used it's possible to opt-out its usage by setting `freshness: 0` in the action being dispatched. Opt-out of the mechanism is not recommended unless in the presence of a specific case where fresh data is really needed.
+
+The default value for the freshness of the actions of a given type is set in the handler so if the action does not set its own freshness value the one set in the handler of the request is used. There is no global default value if the both handler and action don't set a freshness value, freshness is not used.
+
+In order to distinguish one request from another, the data-layer uses information of all the properties in the request action excluding meta and freshness properties. So if two request actions have all properties equal they are assumed to be the same request. If there is a difference in the properties, they are assumed to be different requests. Internally the data-layer converts the action to a key (string) and then uses that keys to compare requests. 
+
+## A possible polling implementation
+
+Polling is not recommended in calypso and should only be used in the presence of a very specific situation where other approaches are not possible. e.g. keeping data updated that comes from an external service we don't control and no other update mechanism is possible.
+In the presence of such a special case freshness mechanism mixed with the [interval component](https://github.com/Automattic/wp-calypso/tree/master/client/lib/interval) allows a simple polling solution.
+
+The polling solution consists in adding the interval component to the render function of the query component where we need periodic requests. The interval component is used to periodically dispatch an action request with the freshness needed. The freshness mechanism avoids the number of requests being multiplied by the number of query components on the page because duplicate requests are removed.
+If there exists more than one polling component and one component is unmount the polling continues because the other component continues to issue requests.
+It is possible to safely add a "polling" query component to the places that need that fresh data. When the data is not needed by any component the requests will stop being executed.
```

Comment:
How about an example of a  basic query component implementation with polling?

I don't think we need to warn so strongly against polling here since we don't have great alternatives at the moment.